diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 74b0423..68133ca 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -235,6 +235,8 @@ endif(CLEAR_MEMORY)
 add_feature_info("Shared library" BUILD_SHARED_LIBS
   "creating libssh2 as a shared library (.so/.dll)")
 
+cmake_policy(SET CMP0074 NEW)
+set(ZLIB_ROOT ../../zlib/ZlibBin)
 option(ENABLE_ZLIB_COMPRESSION "Use zlib for compression")
 add_feature_info(Compression ENABLE_ZLIB_COMPRESSION
   "using zlib for compression")
